### PR TITLE
Mccalluc/sniff zarr for azimuth

### PR DESCRIPTION
**Pull request by mccalluc**
- If this goes ahead, this will replace https://github.com/hubmapconsortium/portal-ui/pull/2313
- If integrated into portal-ui, it would fix https://github.com/hubmapconsortium/portal-ui/issues/2298

... but right now, it's just a reproducer for the problem with stalled requests:
```
export GROUPS_TOKEN='paste-your-token'
export ASSETS_URL='https://assets.hubmapconsortium.org'
PYTHONPATH=. pytest . -vv -k 503f9c
================================================== test session starts ==================================================
platform darwin -- Python 3.7.11, pytest-5.2.1, py-1.11.0, pluggy-0.13.1 -- /opt/anaconda3/envs/portal-vis/bin/python
cachedir: .pytest_cache
rootdir: /Users/chuck/github/hubmap/portal-visualization
collected 11 items / 10 deselected / 1 selected

test/test_builders.py::test_entity_to_vitessce_conf[RNASeqAnnDataZarrViewConfBuilder/503f9caeffb9ee1e1324793c977d2342-entity.json] ^C
```
and it just hangs.

- Ilan has run a local http server containing the zarr, and has done logging from the server side.
- I have not done that, yet: I have tried to log requests from the client side, but because it's using async http under the hood this is more complicated. I've tried a little bit inside python, and a little bit from outside (`httpry`), but haven't gotten anything from either approach.

### CODE REVIEW

**Comment by yuanzhou** [comment]
It has nothing to do with the zarr library nor the SSL. The request of sub-directory
`https://assets.hubmapconsortium.org/44830922b634a9ecd4812b4f27ad4923/hubmap_ui/anndata-zarr/secondary_analysis.zarr/uns/annotation_metadata/is_annotated/`
returns. And you **MUST use the trailing slash `/`** since it's a directory. 

Requesting the actual file returns the file content or the downloadable:
`https://assets.hubmapconsortium.org/44830922b634a9ecd4812b4f27ad4923/hubmap_ui/anndata-zarr/secondary_analysis.zarr/uns/annotation_metadata/is_annotated/.zarray`

And if the requested file or directory doesn't exist, it returns 404. But don't forget the trailing slash `/` for directories.
`https://assets.hubmapconsortium.org/44830922b634a9ecd4812b4f27ad4923/hubmap_ui/anndata-zarr/secondary_analysis.zarr/uns/annotation_metadata/test_dir/`

**Comment by mccalluc** [comment]
When I reference this version in my local portal instance and visit http://localhost:5001/browse/dataset/503f9caeffb9ee1e1324793c977d2342, I see
Screen Shot 2022-02-15 at 11 37 24 AM

Anything else worth checking?

**Comment by ilan-gold** [comment]
@mccalluc Does http://localhost:5001/browse/dataset/7cc96210e80d5736ca38711503e4ff06 work for you on PROD?

File: src/builders/anndata_builders.py [hunk 1]
```diff
@@ -34,11 +36,20 @@ def get_conf_cells(self):
...
+        dags = [
+            dag for dag in self._entity['metadata']['dag_provenance_list']
+            if 'name' in dag]
         if(any(['azimuth-annotate' in dag['origin'] for dag in dags])):
-            cell_set_obs.append("predicted.ASCT.celltype")
-            cell_set_obs_names.append("Predicted ASCT Cell Type")
+            response = requests.get(
+                f'{adata_url}/uns/annotation_metadata/is_annotated/0',
+                headers=self._get_request_init()['headers'])
+            if response.ok:
```
Comment (ilan-gold) [inline src/builders/anndata_builders.py:46]: We need to check the actual value returned.  This isn't just about its presence but whether or not the returned bytes are 1 or 0

Comment (ilan-gold) [reply src/builders/anndata_builders.py:46]: So checking `response.content == b'\x01'` is probably what we want

**Review by ilan-gold — CHANGES_REQUESTED** [review]

File: src/builders/anndata_builders.py (hunk 1 above)
Comment (mccalluc) [reply src/builders/anndata_builders.py]: will do: thanks!

**Review by ilan-gold — CHANGES_REQUESTED** [review]
http://localhost:5001/browse/dataset/44830922b634a9ecd4812b4f27ad4923 seems to be erroring...`_get_request_init` can return `None` if the dataset is published

**Comment by mccalluc** [comment]
@ilan-gold -- Good catch! Sorry to miss that.

**Comment by yuanzhou** [comment]
For documentation purpose. We've concluded the discussion and agreed upon changes on the zarr directory/file handling via the File Assets Service:
- The slowness of reported request is not due to neither the zarr library nor the SSL. It's because the File Assets Nginx requires a trailing slash if the target path is a directory in URL, otherwise it does 301 redirect by default behavior.
- We agreed to handle the directory in URL path (with or without trailing slash) by returning 404 for both cases. And we've verified on PROD with the quick tweak I made. I'll need to push/release the changes and redeploy later on all environments. 
- If a file being requested in the assets URL, 200/401/403/404/500 are the expected response status codes. It's been working like this.
- For everyone to understand it better, the File Assets Service is not a REST API. The Gateway handles the auth check just like all other APIs (entity-api, search-api, etc.), but once the auth passed, it's a PURE Nginx that handles the static files stored on the PSC machines.

File: src/builders/anndata_builders.py [hunk 2]
```diff
@@ -34,11 +36,18 @@ def get_conf_cells(self):
...
+            if 'name' in dag]
         if(any(['azimuth-annotate' in dag['origin'] for dag in dags])):
-            cell_set_obs.append("predicted.ASCT.celltype")
-            cell_set_obs_names.append("Predicted ASCT Cell Type")
+            request_init = self._get_request_init() or {}
+            headers = request_init.get('headers', {})
+            response = requests.get(
+                f'{adata_url}/uns/annotation_metadata/is_annotated/0',
+                headers=headers)
+            if response.content == b'\x01':
```
Comment (john-conroy) [inline src/builders/anndata_builders.py:48]: I share your confusion, Chuck. Maybe add a comment to explain the significance of `b'\x01'`?

**Review by john-conroy — APPROVED** [review]

File: src/builders/anndata_builders.py (hunk 2 above)
Comment (mccalluc) [reply src/builders/anndata_builders.py:48]: I'll file an issue. Thanks for the nudge.

**Pull request merged**